### PR TITLE
Added guide for CycleGAN

### PR DESCRIPTION
**Pull request by mayank2**
\<insert pull request description here\>

-----

# Checklist

Please go through each entry in the below checklist and mark an 'X' if that condition has been met. Every entry should be marked with an 'X' to be get the Pull Request approved.

- [ ] All `import`s are in the first cell? First block of imports are standard libraries, second block are 3rd party libraries, third block are all `arcgis` imports? Note that in some cases, for samples, it is a good idea to keep the imports next to where they are used, particularly for uncommonly used features that we want to highlight.
- [ ] All `GIS` object instantiations are one of the following?
    - `gis = GIS()`
    - `gis = GIS('https://www.arcgis.com', 'arcgis_python', 'P@ssword123')`
    - `gis = GIS(profile="your_online_profile")`
    - `gis = GIS('https://pythonapi.playground.esri.com/portal', 'arcgis_python', 'amazing_arcgis_123')`
    - `gis = GIS(profile="your_enterprise_portal")`
- [ ] If this notebook requires setup or teardown, did you add the appropriate code to `./misc/setup.py` and/or `./misc/teardown.py`?
- [ ] If this notebook references any portal items that need to be staged on AGOL/Python API playground, did you coordinate with a Python API team member to stage the item the correct way with the api\_data\_owner user?
- [ ] Code refactored & split out across multiple cells, useful comments?
- [ ] Consistent voice/tense/narrative style? Thoroughly checked for typos?
- [ ] All images used like `` instead of ``? All map widgets contain a static image preview? (Call `mapview_inst.take_screenshot()` to do so)
- [ ] All file paths are constructed in an OS-agnostic fashion with `os.path.join()`? (Instead of `r"\foo\bar"`, `os.path.join(os.path.sep, "foo", "bar")`, etc.)
- [ ] **IF YOU WANT THIS SAMPLE TO BE DISPLAYED ON THE DEVELOPERS.ARCGIS.COM WEBSITE**, ping @ DavidJVitale so he can add it to the list for the next deploy

### CODE REVIEW

**Comment by priyankatuteja** [comment]
@sdash77 Please do the first round of reviews.

File: guide/14-deep-learning/How CycleGAN Works.ipynb [hunk 1]
```diff
@@ -0,0 +1,128 @@
+{
```
Comment (sdash77) [inline guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <p>could start by describing what type of model is cyclegan or about image-to-image translation, then can continue "There are situations.......".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (sdash77) [inline guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <p>Please add a reference for the image in the references section</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (sdash77) [inline guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <p>which guide ?, no reference</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (sdash77) [inline guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <p>"In <code>arcgis.learn</code>, all the discriminators and generators have been grouped into a single model" Please remove this, it is the inherent architecture of cyclegan. Its grouped everywhere.</p><p> </p><p>Describe a little about the objective function or loss that we get combining the three losses discussed above. feels incomplete.</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (sdash77) [inline guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <p>add reference for model architecture image</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (mayank2) [reply guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <p>Reference cannot be linked until Pix2Pix has been merged.</p>

**Comment by sdash77** [comment]
looks fine to me @mayank2 Thanks.

File: guide/14-deep-learning/How CycleGAN Works.ipynb [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
```
Comment (sdash77) [inline guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <p>Figure 1. Overview of CycleGAN architecture <code>translating from image to map domain</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (sdash77) [inline guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <p>It captures the intuition that if we translate the image from one domain to the other and back again we should arrive at where we started. Hence, it calculates the L1 loss between the original image and the final generated image, which should look same as original image. <code>It is calculated in two directions:</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

**Comment by sdash77** [comment]
Seems fine to me.

**Review by sdash77 — APPROVED** [review]

File: guide/14-deep-learning/How CycleGAN Works.ipynb (hunk 2 above)
Comment (priyankatuteja) [inline guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <p>Provide a reference link to the paper for example:</p><p>Figure 1. Overview of CycleGAN architecture: Translating from satellite image to map routes domain [1]</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (priyankatuteja) [inline guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <blockquote>Each generator has a corresponding discriminator model. The first discriminator model (Discriminator-A) takes real images from Domain-A and generated images from Generator-A and predicts whether they are real or fake. The second discriminator model (Discriminator-B) takes real images from Domain-B and generated images from Generator-B and predicts whether they are real or fake. </blockquote><p>Each generator has a corresponding discriminator model. The discriminator model takes real images from Domain and generated images from Generator to predict whether they are real or fake.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

File: guide/14-deep-learning/How CycleGAN Works.ipynb [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
```
Comment (AtmaMani) [inline guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <p>Change filename to 'how_cyclegan_works.ipynb' (no spaces, all lowercase)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (AtmaMani) [inline guide/14-deep-learning/How CycleGAN Works.ipynb:1]: <p>"There are some challenges in Pix2Pix is that the images should be paired which are randomly organized or unpaired and we want to convert images from one domain to another just like " -&gt; <strong>this sentence seems incomplete. Further, can you elaborate or reword the limitations of Pix2Pix model?</strong></p><p> </p><p><strong>I think we also need to elaborate what you mean by an unpaired dataset</strong></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Review by AtmaMani — CHANGES_REQUESTED** [review]
Changes requested via reviewnb

File: guide/14-deep-learning/How CycleGAN Works.ipynb (hunk 3 above)
Comment (mayank2) [reply guide/14-deep-learning/How CycleGAN Works.ipynb]: Yes, the sentence is incomplete. Thanks for pointing it out .

**Review by AtmaMani — APPROVED** [review]

**Pull request merged**